packages/flare/tests/bin/test_ingest_events.py

```diff
@@ -2,287 +2,375 @@
-    app = MagicMock()
-    for _ in fetch_feed(
-        logger=logger,
-        app=app,
-        api_key="some_key",
-        tenant_id=11111,
-        ingest_metadata_only=False,
-    ):
-        pass
-
-    logger.error.assert_called_once_with("Exception=Failed to fetch API Token")
-
-
-@patch("cron_job_ingest_events.FlareAPI")
-@patch("time.sleep", return_value=None)
-def test_fetch_feed_expect_feed_response(
-    sleep: Any, flare_api_mock: MagicMock, capfd: Any
-) -> None:
-    logger = MagicMock()
-    app = MagicMock()
-
-    next = "some_next_value"
-    first_item = {
-        "actor": "this guy",
-    }
-    second_item = {
-        "actor": "some other guy",
-    }
-    expected_items = [first_item, second_item]
-    flare_api_mock_instance = flare_api_mock.return_value
-    flare_api_mock_instance.fetch_feed_events.return_value = iter(
-        [(first_item, next), (second_item, next)]
-    )
-
-    events: list[dict] = []
-    for event, next_token in fetch_feed(
-        logger=logger,
-        app=app,
-        api_key="some_key",
-        tenant_id=11111,
-        ingest_metadata_only=False,
-    ):
-        assert next_token == next
-        events.append(event)
-
-    for i in range(len(events)):
-        assert events[i] == expected_items[i]
-
-
-@patch(
-    "cron_job_ingest_events.get_last_fetched",
-    return_value=datetime.now() - timedelta(minutes=5),
-)
-def test_main_expect_early_return(get_last_fetched_mock: MagicMock) -> None:
-    logger = MagicMock()
-    app = MagicMock()
-
-    main(logger=logger, app=app)
-    logger.info.assert_called_once_with(
-        f"Fetched events less than {int(CRON_JOB_THRESHOLD_SINCE_LAST_FETCH.seconds / 60)} minutes ago, exiting"
-    )
-
-
-@patch("cron_job_ingest_events.fetch_feed")
-@patch(
-    "cron_job_ingest_events.get_ingest_metadata_only",
-    return_value=(False),
-)
-@patch(
-    "cron_job_ingest_events.get_tenant_id",
-    return_value=(111),
-)
-@patch(
-    "cron_job_ingest_events.get_api_key",
-    return_value=("some_api_key"),
-)
-@patch(
-    "cron_job_ingest_events.get_last_fetched",
-    return_value=datetime.now() - timedelta(minutes=10),
-)
-def test_main_expect_normal_run(
-    get_last_fetched_mock: MagicMock,
-    get_api_key_mock: MagicMock,
-    get_tenant_id_mock: MagicMock,
-    get_ingest_metadata_only_mock: MagicMock,
-    fetch_feed_mock: MagicMock,
-) -> None:
-    logger = MagicMock()
-    app = MagicMock()
-
-    main(logger=logger, app=app)
-    fetch_feed_mock.assert_called_once_with(
-        logger=logger,
-        app=app,
-        api_key="some_api_key",
-        tenant_id=111,
-        ingest_metadata_only=False,
-    )
+T = TypeVar("T", bound="StoragePassword")
+
+
+class StoragePassword:
+    def __init__(self, username: str, clear_password: str) -> None:
+        self._state = {
+            "username": username,
+            "clear_password": clear_password,
+        }
+
+    @property
+    def content(self: T) -> T:
```

Comment:
What is the typing trying to do here?

Reply:
Can't type return to Self. This is what the internet told me to do.

Reply:
Ah because `typing.Self` is python > 3.9 ?

Reply:
Without `Self` you can just use `"StoragePassword"` which does not need generics.

You need to keep the quotes, since the class is not yet defined.